llvm/lib/Support/OutputBackend.cpp

```diff
@@ -583,8 +583,8 @@ std::unique_ptr<raw_pwrite_stream> OnDiskOutputFile::takeOSImpl() {
   if (OS->supportsSeeking() || Config.getText())
     return std::move(OS);
 
-  // Fall back on the content buffer.
-  return createStreamForContentBuffer();
+  // Fall back to buffer_unique_ostream.
+  return std::make_unique<buffer_unique_ostream>(std::move(OS));
```

Comment:
I copy that exact line of code from original implementation from `CompilerInstance.cpp`. I don't know the exact problem but `OS` has to be moved here, otherwise PIPE just wont work on OutputBackend.

Reply:
Looks to me like pipes do work:
```
% ./build/bin/clang -E -x c /dev/null | head    
# 1 "/dev/null"
# 1 "<built-in>" 1
# 1 "<built-in>" 3
# 353 "<built-in>" 3
# 1 "<command line>" 1
# 1 "<built-in>" 2
# 1 "/dev/null" 2

% ./build/bin/clang -dM -E -x c /dev/null | head
#define _LP64 1
#define __APPLE_CC__ 6000
#define __APPLE__ 1
#define __ATOMIC_ACQUIRE 2
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_CONSUME 1
#define __ATOMIC_RELAXED 0
#define __ATOMIC_RELEASE 3
#define __ATOMIC_SEQ_CST 5
#define __BIGGEST_ALIGNMENT__ 16
```
Which specific tests are you looking at? I can take over fixing those ones.

Reply:
I was thinking just abandon the output backend parts since we can just merge in the upstream code which might be very different. For failing tests on this branch, you can look at:
`clang/test/Frontend/rewrite-includes-missing.c`

Reply:
Okay, SGTM.

If it's just one test, you could also add `REQUIRES: disabled` with a FIXME saying that output backend breaks it. I seem to remember rewrite-includes being weird.